python/rmm/_lib/device_buffer.pyx

```diff
@@ -285,6 +285,58 @@ cdef class DeviceBuffer:
 
         return b
 
+    cpdef DeviceBuffer join(self, list L, Stream stream=DEFAULT_STREAM):
+        """Joins a sequence of ``DeviceBuffer``s with ``self`` inbetween.
+
+        This is analogous to ``bytes.join(...)``.
+
+        Parameters
+        ----------
+        L : ``list`` of ``DeviceBuffer``s
+        stream : CUDA stream to use for copying, default the default stream
+
+        Examples
+        --------
+        >>> import rmm
+        >>> db = rmm.DeviceBuffer()
+        >>> db1 = rmm.DeviceBuffer.to_device(b"abc")
+        >>> db2 = rmm.DeviceBuffer.to_device(b"def")
+        >>> rdb = db.join([db1, db2])
+        >>> hb = rdb.copy_to_host()
+        >>> print(hb)
+        array([ 97,  98,  99, 100, 101, 102], dtype=uint8)
+        """
+        cdef uintptr_t sp = <uintptr_t>self.c_data()
+        cdef size_t sdbs = self.c_size()
```

Comment:
lots of very short and undescriptive variable names in this function.

Reply:
Definitely could use clearer and more canonical names like `sp->sep`, `sdbs->sep_bytes`, etc.